mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java

```diff
@@ -263,49 +268,189 @@ public Service createMicoServiceInterface(MicoServiceInterface micoServiceInterf
+//   		return MicoApplicationDeploymentStatus.incompleted("Deployment of MicoApplication failed.",
+//   			"Service 1 could not be deployed.", "Service 2 could not be deployed");
+    	
+    	log.debug("Check application deployment status for MicoApplication '{}' '{}'.",
+    		micoApplication.getShortName(), micoApplication.getVersion());
+   		
+   		// Check deployment status based on stored Kubernetes deployment information
+   		List<MicoServiceDeploymentInfo> micoServiceDeploymentInfos = serviceDeploymentInfoRepository.findAllByApplication(
+   			micoApplication.getShortName(), micoApplication.getVersion());
+   		if (micoServiceDeploymentInfos.isEmpty()
+   			|| micoServiceDeploymentInfos.stream().allMatch(sdi -> sdi.getKubernetesDeploymentInfo() == null)) {
+   			// If the MicoApplication does not provide service deployment information, or
+   			// all Kubernetes deployment information is null -> application is currently (intentionally) not deployed
+            log.debug("There are no service deployment information with Kubernetes deployment information " +
+                    "for the MicoApplication '{}' '{}' stored in the database. MicoApplication is considered to be not deployed.",
                 micoApplication.getShortName(), micoApplication.getVersion());
-        }
-
-        String deploymentStatus = result ? "deployed" : "not deployed";
-        log.info("MicoApplication '{}' in version '{}' is {}.",
-            micoApplication.getShortName(), micoApplication.getVersion(), deploymentStatus);
-        return result;
+   			return MicoApplicationDeploymentStatus.undeployed("MicoApplication is currently not deployed.");
+   		}
+   		
+    	// Default application deployment status is 'deployed'. The following code will
+   		// use a strategy which checks whether there are errors or inconsistencies within
+   		// the stored deployment information and the actual Kubernetes deployments and services.
+   		// If there are such errors or inconsistencies, this variable will be updated.
+   		MicoApplicationDeploymentStatus.Value applicationDeploymentStatus = Value.DEPLOYED;
+   		List<MicoMessage> messages = new ArrayList<>();
+   		String msg;
+   		
+   		// Check whether there are jobs currently running for the deployment of the MicoApplication
+   		MicoApplicationJobStatus applicationJobStatus = backgroundJobBroker
+		    .getJobStatusByApplicationShortNameAndVersion(micoApplication.getShortName(), micoApplication.getVersion());
+   		switch (applicationJobStatus.getStatus()) {
+   		case PENDING:
+   			// 'Pending' indicates that the deployment is scheduled to be executed in the future
+   			applicationDeploymentStatus = Value.PENDING;
+   			msg = "The deployment of MicoApplication '" + micoApplication.getShortName()
+                + "' '" + micoApplication.getVersion() + "' is scheduled to be started.";
+   			log.debug(msg);
+   			messages.add(MicoMessage.error(msg));
```

Comment:
Currently for all states of the deployment `MicoMessage.error` is used. What about `info` and `warning`?
I think `PENDING` is no error, it is a information (or at max a warning).

Reply:
Agree, i suggest the following mapping:
 - `PENDING`, `RUNNING`, `DEPLOYED`, `UNDEPLOYED` -> `INFO`
 - `UNKNOWN -> `WARNING`
 - `INCOMPLETE`, _failed background job for deployment_ -> `ERROR`

Reply:
Sounds good 👍